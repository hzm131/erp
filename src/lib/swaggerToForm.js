import _map from 'lodash/map'

export default function swaggerToForm(obj, fields) {
  return _map(obj.properties, (v, k) => {
    v.required = obj.required && obj.required.includes(k)

    fields = obj.fields || fields;

    if(fields){
      v.fields = {}

      v.fields.create = fields.create.includes(k)
      v.fields.edit = fields.edit.includes(k)
      v.fields.list = fields.list.includes(k)
      v.fields.detail = fields.detail.includes(k)
    }

    const rules = []
    if (v.required) {
      rules.push({
        required: true,
        message: `必须输入${v.description}`
      })
    }

    return {
      name: v.description,
      type: v.type,
      field: k,
      col: {
        span: 12
      },
      rules,
      required: v.required,
      fields: v.fields
    }
  }).filter(v => v)
}
