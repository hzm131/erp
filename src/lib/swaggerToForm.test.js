import swaggerToForm from './swaggerToForm'

const obj = {
  type: 'object',
  fields: {
    edit: ['name'],
    list: ['name'],
    detail: ['name'],
    create: ['name']
  },
  required: ['name'],
  properties: {
    id: {
      type: 'integer'
    },
    name: {
      type: 'string',
      description: '机构名称'
    },
    abbreviation: {
      type: 'string',
      description: '机构简称'
    }
  }
}

it('should render swagger definition to form', () => {
  expect(swaggerToForm(obj)).toMatchSnapshot()
})
