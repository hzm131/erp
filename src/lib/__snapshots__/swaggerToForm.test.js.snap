// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render swagger definition to form 1`] = `
Array [
  Object {
    "col": Object {
      "span": 12,
    },
    "field": "id",
    "fields": Object {
      "create": false,
      "detail": false,
      "edit": false,
      "list": false,
    },
    "name": undefined,
    "required": false,
    "rules": Array [],
    "type": "integer",
  },
  Object {
    "col": Object {
      "span": 12,
    },
    "field": "name",
    "fields": Object {
      "create": true,
      "detail": true,
      "edit": true,
      "list": true,
    },
    "name": "机构名称",
    "required": true,
    "rules": Array [
      Object {
        "message": "必须输入机构名称",
        "required": true,
      },
    ],
    "type": "string",
  },
  Object {
    "col": Object {
      "span": 12,
    },
    "field": "abbreviation",
    "fields": Object {
      "create": false,
      "detail": false,
      "edit": false,
      "list": false,
    },
    "name": "机构简称",
    "required": false,
    "rules": Array [],
    "type": "string",
  },
]
`;
